### PR TITLE
Add support for new informative error messages

### PR DESCRIPTION
**Pull request by kirre-bylund**
Also refactored a lot of error handling in the code.

### CODE REVIEW

**Review by AlmightyMikkel — APPROVED** [review]

**Review by JohannesLoot — APPROVED** [review]
Awesome work 👍 
Great clean up!

File: Runtime/Client/LootLockerBaseServerAPI.cs [hunk 1]
```diff
@@ -97,7 +107,7 @@ IEnumerator coroutine()
                     if (!webRequest.isDone && timedOut)
                     {
                         LootLockerLogger.GetForLogLevel(LootLockerLogger.LogLevel.Warning)("Exceeded maxTimeOut waiting for a response from " + request.httpMethod + " " + url);
-                        OnServerResponse?.Invoke(new LootLockerResponse() { hasError = true, statusCode = 408, text = "{\"error\": \"" + request.endpoint + " Timed out.\"}", Error = request.endpoint + " Timed out." });
+                        OnServerResponse?.Invoke(LootLockerResponseFactory.Error<LootLockerResponse>(request.endpoint + " Timed out."));
```
Comment (kirre-bylund) [inline Runtime/Client/LootLockerBaseServerAPI.cs:110]: We lost the status code here.

File: Runtime/Client/LootLockerBaseServerAPI.cs [hunk 2]
```diff
@@ -170,23 +142,75 @@ IEnumerator coroutine()
...
+                                    case 503:
+                                        errorData.message = "Service Unavailable -- We're either offline for maintenance, or an error that should be solvable by calling again later was triggered.";
+                                        errorData.code = "service_unavailable";
+                                        break;
+                                    default:
+                                        errorData.message = "Unknown error.";
+                                        break;
+                                }
+                            }
+                            errorData.message += " " + ObfuscateJsonStringForLogging(webRequest.downloadHandler.text);
```
Comment (kirre-bylund) [inline Runtime/Client/LootLockerBaseServerAPI.cs:210]: Not sure this is right. In the case of the informative error messages we don't want the whole body after the message. And will there ever be info needing obfuscation in the returned message.

**Review by kirre-bylund — COMMENTED** [review]
This needs to be changed.

**Pull request merged**